tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,689 @@
+
+Note: The script only needs to cover mgmt-plane.
+
+**Example invocations:**
+
+```powershell
+update-changelog.ps1 \
+      -SdkRepoPath C:\dev\repos\azure-sdk-for-net \
+      -PackagePath C:\dev\repos\azure-sdk-for-net\services\newservice
+```
+
+#### 2. Version Update Tool
+
+Name (CLI): `azsdk package update version`
+
+Name (MCP): `azsdk_package_update_version`
+
+Purpose: Increment and propagate package version across all authoritative version surfaces (e.g., `pom.xml`, `_version.py`, `*.csproj`, `package.json`, `go.mod`, `CHANGELOG.md`).
+
+Inputs:
+
+- `--package-path <abs path>` (required)
+- `--release-type <stable|beta>` (optional)
+- `--version` (optional)
+- `--release-date` (optional)
+
+Execution Steps:
+
+1. Compute a version based on the given information (mgmt-plane).
+2. Apply updates across language-specific files.
+3. Validate consistency (all files now share identical new version).
+4. For data-plane outside release: return `noop`.
+5. For data-plane prepare release: use current version from the metadata file or from user input (support may be post-milestone-1).
+
+> **Note:** In milestone 1, Java management-plane packages will follow the same version update strategy as data-plane packages. Adjustments for Java-specific workflows may be introduced in later versions.
+
+Outputs:
+
+```json
+{"result":"succeeded", "version": "1.2.0", "release-date": "2025-10-17", "message":"Version updated 1.2.0-beta.2 -> 1.2.0", "next_steps": "update metadata"}
+```
+
+```json
+{"result":"noop", "version": "", "release-date": "", "message":"Data-plane pre-release context; no version bump performed", "next_steps": "update metadata"}
+```
+
+Failure Modes:
+
+- Version parse error (non-semver) → fail with offending token.
+- Partial update (one file locked) → `partial` list and errors.
+
+Idempotency: Re-running after success yields `succeeded` with `no changes` message when already at target version.
+
+##### Update Version script
+
+When the tool executes the language-specific `update-version` script it will pass the following parameters. All paths are absolute.
+
+| Parameter | Description | Type | Required |
+|-----------|-------------|------:|:--------:|
+| `SdkRepoPath` | Absolute path to the root folder of the local SDK repository. | string | Yes |
+| `PackagePath` | Absolute path to the root folder of the local SDK project (package). | string | Yes |
+| `SdkReleaseType` | Specifies whether the next version is `beta` or `stable`. | string | No |
+| `SdkVersion` | Specifies the next version to set (explicit version string). | string | No |
+| `SdkReleaseDate` | Release date to write into changelog (YYYY-MM-DD). | string | No |
+
+Note: For management-plane (mgmt) packages, at least one of `SdkVersion` or `SdkReleaseType` MUST be provided; the script will fail with an error if neither is supplied or validation fails.
+
+**Example invocations:**
+
+```powershell
+update-version.ps1 \
```

Comment:
For data plane is this doing anything more than wrapping the prepare release script? Does it do more in mgmt plane? It seems unnecessary to define a new script

Reply:
This script is for mgmt-plane only. For data-plane, I'll check whether we can call the `prepare-release` script directly or any adjustments are needed after m1.